examples/api/lib/material/selection_area/selection_area.1.dart

```diff
@@ -0,0 +1,202 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math';
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+
+/// Flutter code sample for [SelectionArea].
+
+void main() => runApp(const SelectionAreaComplexEmphasizeTextExampleApp());
+
+class SelectionAreaComplexEmphasizeTextExampleApp extends StatelessWidget {
+  const SelectionAreaComplexEmphasizeTextExampleApp({super.key});
+
+  // This widget is the root of your application.
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Flutter Demo',
+      theme: ThemeData(
+        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
+        useMaterial3: true,
+      ),
+      home: const MyHomePage(title: 'Flutter Demo Home Page'),
+    );
+  }
+}
+
+class MyHomePage extends StatefulWidget {
+  const MyHomePage({super.key, required this.title});
+
+  final String title;
+
+  @override
+  State<MyHomePage> createState() => _MyHomePageState();
+}
+
+class _MyHomePageState extends State<MyHomePage> {
+  final ContextMenuController _menuController = ContextMenuController();
+  final SelectionController _selectionController = SelectionController();
+
+  @override
+  void dispose() {
+    _selectionController.dispose();
+    super.dispose();
+  }
+
+  String _trimPlaceholders(String text) {
+    return text.replaceAll(String.fromCharCode(PlaceholderSpan.placeholderCodeUnit), '');
+  }
+
+  void _emphasizeText(List<SelectedContentController<Object>>? controllers) {
```

Comment:
This method seems quite complex, I hope we can have a  more generic api to modify textspan. In android, you can do something like SpannableString.setSpan(start, end, certainAttribute).